tls/s2n_recv.c

```diff
@@ -205,7 +205,7 @@ ssize_t s2n_recv_impl(struct s2n_connection *conn, void *buf, ssize_t size, s2n_
             continue;
         }
 
-        out.size = MIN(size, s2n_stuffer_data_available(&conn->in));
+        out.size = MIN((uint32_t) size, s2n_stuffer_data_available(&conn->in));
```

Comment:
Same here - it might be good to check that the user passed in a positive value for `size` before casting.